mono/metadata/icall.c

```diff
@@ -888,7 +888,12 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_InitializeArray (MonoAr
 		return;
 	}
 
-	if (!(field_handle->type->attrs & FIELD_ATTRIBUTE_HAS_FIELD_RVA)) {
+
+	MonoType *field_type = mono_field_get_type_checked (field_handle, error);
+	if (!field_type)
```

Comment:
Shouldn't this check for `error` instead (as well) ?

Reply:
No, functions that have ternary return state are clearly labeled so. IE, if `field_type == NULL` error must be set.